ui/src/pages/ChallengePage.js

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useEffect } from "react";
+import "../style/ChallengePage.css";
+import * as core from "@material-ui/core";
+
+import PropTypes from "prop-types";
+import Terminal from "../components/Terminal";
+
+import gfm from "remark-gfm";
+import ReactMarkdown from "react-markdown";
+import useFetchAuth from "../useFetchAuth";
+import Spinner from "../components/Spinner";
+import GenericErrorPage from "./error-pages/genericErrorPage";
+import {useParams} from "react-router-dom";
+
+function a11yProps(index) {
+	return {
+		id: `scrollable-auto-tab-${index}`,
+		"aria-controls": `scrollable-auto-tabpanel-${index}`,
+	};
+}
+
+function TabPanel(props) {
+	const { children, value, index, ...other } = props;
+	return (
+		<div
+		role="tabpanel"
+		hidden={value !== index}
+		id={`scrollable-auto-tabpanel-${index}`}
+		aria-labelledby={`scrollable-auto-tab-${index}`}
+		{...other}
+		>
+		<core.Box style={{ padding: 0 }} p={3}>
+			<core.Typography>{children}</core.Typography>
+		</core.Box>
+		</div>
+	);
+}
+
+TabPanel.propTypes = {
+	children: PropTypes.node,
+	index: PropTypes.any.isRequired,
+	value: PropTypes.any.isRequired,
+};	
+
+export default function ChallengePage() {
+	//value variable control which is the chosen/selected tab in appbar
+	const [value, setValue] = React.useState(0);
+	const handleChange = (event, newValue) => {
+		setValue(newValue);
+		const temp=`/api/challenge-sets/${csSlug}/challenges/${cSlug}`.toString()+'/docs/'
+		+fetchData.documentation[newValue].path;
+		// console.log(temp);
+		setPath(temp);
+	};
+
+	const [txt, setMarkdown] = useState("");
+	const [path, setPath] = useState("");
+	const [fetchData, setData] = useState(0);
+
+
+	let {csSlug, cSlug} = useParams();
+
+	// console.log('testing');
+	// console.log(`${csSlug}-${cSlug}`);
+	useEffect(() => {
+		fetch(`/api/challenge-sets/${csSlug}/challenges/${cSlug}`)
+			.then((res) => res.json())
+			.then((json) => {
+							// console.log(json);
+							setValue(0);
+							setData(json);
+							const temp=`/api/challenge-sets/${csSlug}/challenges/${cSlug}`.toString()
+							+'/docs/'+json.documentation[0].path;
+							// console.log(temp);
+							setPath(temp)	
+						});
+	}, [csSlug, cSlug]);
+
+	//Testing git commit for automatically add initials
+
+	// this useEffect gets triggered when path hook is updated
+	useEffect(() => {
+		fetch(path)
+		.then((res) => res.text())
+		.then((text) => {
+			setMarkdown(text);
+		});
+	}, [path]);
+
+	// if (loading) return <Spinner />;
```

Comment:
why is this commented out?

Reply:
The next 3 lines are commented out because we are not using useAuthFetch to get data any more. I used useEffect and then fetch to retrieve the data. Without using useAuthFetch, we cannot have [loading, error, data]. Unless i am wrong about these

Reply:
Most updated version should have Spinner, ErrorPage, and Terminal recovered